src/Compiler/Common.idr

```diff
@@ -180,8 +180,7 @@ natHackNames
     = [UN "prim__add_Integer",
        UN "prim__sub_Integer",
        UN "prim__mul_Integer",
-       NS preludeNS (UN "natToInteger"),
```

Comment:
Did you mean to remove `natToInteger` here? Sorry, not familiar with the machinery here yet but not sure how replacing `integerToNat` would remove the need for the inverse here and I noticed `natToInteger` is still handled in `natHack`

Reply:
Yep: `natToInteger` is not used (it is removed by [this magic](https://github.com/idris-lang/Idris2/pull/799/files#diff-0fc453ae86601d6a1ec44d86a6d4c4c4d6d23ac8e74ab78797931f4bc8dff852R174)) so there is no point
in making sure it is part of the generated code. My guess is that this got out of sync
a long time ago.

I only noticed it because my `prim_integerToNat` was not getting compiled. So I
started looking for the code that computes a definition's dependencies to decide
what to include in the executable and found this.

Reply:
Ohhh. Yeah. That makes sense. I had it backwards in my head; it’s actually that `natHack` begets `natHackNames`. Thanks for the explanation!

Reply:
Thanks for the review!